### DISCUSSION TITLE
Cannot find module 'swiper/css' or its corresponding type declarations. ts(2307)

### DISCUSSION THREAD

**Discussion by jmaldon1** [nolimits4web/swiper#6103] [Q&A]
I am using [Remix ](https://github.com/remix-run/remix) and in order to import css files in remix you must do it like this:

```typescript
import swiperCss from "swiper/css";  // ERROR: Cannot find module 'swiper/css' or its corresponding type declarations. ts(2307)

export const links: LinksFunction = () => {
  return [
    {
      rel: "stylesheet",
      // href: "https://unpkg.com/swiper@8/swiper-bundle.min.css",
      href: swiperCss,
    },
  ];
};
```

Does anyone else have a way to handle this without doing `@ts-expect-error` to ignore the error?

**Comment by MangoMarcus** [comment]
I have the same issue, although only when using dynamic imports eg. `import('swiper/css')`

My only fix is to declare a types file

```ts
// swiper.d.ts
declare module 'swiper/css' {}
declare module 'swiper/css/pagination' {}
declare module 'swiper/css/autoplay' {}
declare module 'swiper/css/effect-fade' {}
declare module 'swiper/css/navigation' {}
// Any other styles you need...
```

**Reply by HT-35** [reply]
> Tôi cũng gặp vấn đề tương tự, mặc dù chỉ khi sử dụng lệnh nhập động, ví dụ:`import('swiper/css')`
> 
> Cách sửa chữa duy nhất của tôi là khai báo một tệp kiểu
> 
> ```ts
> // swiper.d.ts
> declare module 'swiper/css' {}
> declare module 'swiper/css/pagination' {}
> declare module 'swiper/css/autoplay' {}
> declare module 'swiper/css/effect-fade' {}
> declare module 'swiper/css/navigation' {}
> // Any other styles you need...
> ```

tks

**Comment by Rick-Phoenix** [comment]
Another solution is to import the css modules from a .css file instead of a JSX/TSX file:
```
@import "swiper/css/pagination";
@import "swiper/css/scrollbar";
@import "swiper/css";
@import "swiper/css/effect-coverflow";
```

**Reply by kelvinmati** [reply]
This worked for me.
Thank you very much

**Reply by asliddinrashidovich** [reply]
this worked for me. Thanks

**Reply by behroodrzdr** [reply]
this alos works for me by creating a sperated css file(swiper.css) and then add import's into.

**Reply by mimiicodes** [reply]
This worked ike magic. Thankss!

**Reply by eabxrry** [reply]
thanks bruh

**Comment by HarshJoshi09-cyber** [comment]
Thanks Bro This solved my issue..

**Comment by HarshJoshi09-cyber** [comment]
Hey Can Anyone have an idea of How to resolve the error i'm getting stucked at please help me find out the Answer
```

Property 'default' does not exist on type 'typeof Slider'.

152                     <Slider.default className='feature-book' {...sliderSettings}>

```
in react Slick slider for 

`            <div className="ml-auto mr-auto w-full flex flex-col justify-center items-center pb-20 relative! z-100!">

                <div className=" w-355 grid grid-cols-1 justify-center items-center  gap-5   p-4 border-b border-b-[#E0E0E0] " >

                    <Slider.default className='feature-book' {...sliderSettings}>

                        {FeaturedBooksData}

                    </Slider.default>

                </div>
                <div className="w-355 flex justify-end p-10">

                    <button className="flex items-center gap-2 text-[#111111] text-[16px] font-medium font-plus transition delay-50 duration-200 ease-in-out hover:-translate-y-1 hover:scale-110 hover:bg-[#74642F] hover:text-white p-2" onClick={() => navigate('/books')}>View All Products <IoIosArrowRoundForward size={25} /></button>
                </div>
            </div>`
